source/client/options_impl.cc

```diff
@@ -671,7 +671,14 @@ CommandLineOptionsPtr OptionsImpl::toCommandLineOptionsInternal() const {
       request_options->mutable_request_body_size()->set_value(requestBodySize());
     }
   }
-  *(command_line_options->mutable_tls_context()) = tls_context_;
+
+  // Only set the tls context if it looks non-empty, to avoid a warning being logged about field
+  // deprecation. Ideally this would follow the way transport_socket uses absl::optional below.
+  // But as this field is about to get eliminated this minimal effort shortcut may be more suitable.
+  if (!Envoy::MessageUtil()(tls_context_,
```

Comment:
Can we add a comment explaining how we are using Envoy::MessageUtil to verify that the tls_context_ message is non-empty?

(optional) On another note - isn't there a more direct way of checking if it is non-empty? E.g. proto2 used to have a method that could be used for that - my_proto.ByteSizeLong() == 0. However that call tends to be very inefficient on messages with many nested fields and I am unsure if it works in proto3 also.

Reply:
I found that theres also `Envoy::Protobuf::util::MessageDifferencer::Equivalent` these days, which I think is easier to grok. See 6b8a0889c10225a3f036c8c098571832f03b900c, does that look better?